osmomath/decimal_test.go

```diff
@@ -739,3 +739,263 @@ func (s *decimalTestSuite) TestLog2() {
+			initialValue:  NewBigDec(0),
+			base:          NewBigDec(2),
+			expectedPanic: true,
+		},
+		"log_(-1)(2): invalid base, normal argument - panics": {
+			initialValue:  NewBigDec(2),
+			base:          NewBigDec(-1),
+			expectedPanic: true,
+		},
+		"log_1(2): base cannot equal to 1 - panics": {
+			initialValue:  NewBigDec(2),
+			base:          NewBigDec(1),
+			expectedPanic: true,
+		},
+		"log_30(100) = 1.353984985057691049642502891262784015": {
+			initialValue: NewBigDec(100),
+			base:         NewBigDec(30),
+			// From: https://www.wolframalpha.com/input?i=log_30%28100%29+to+37+digits
+			expected: MustNewDecFromStr("1.353984985057691049642502891262784015"),
+		},
+		"log_0.2(0.99) = 0.006244624769837438271878639001855450": {
+			initialValue: MustNewDecFromStr("0.99"),
+			base:         MustNewDecFromStr("0.2"),
+			// From: https://www.wolframalpha.com/input?i=log_0.2%280.99%29+to+34+digits
+			expected: MustNewDecFromStr("0.006244624769837438271878639001855450"),
+		},
+		// THIS TEST CASE FAILS WITH 0.722861284485807159166346369373275313 ERROR
+		// "log_0.0001(500000) = -1.424742501084004701196565276318876743": {
+		// 	initialValue: NewDecWithPrec(1, 4),
+		// 	base:         NewBigDec(500000),
+		// 	// From: https://www.wolframalpha.com/input?i=log_0.0001%28500000%29+to+37+digits
+		// 	expected: MustNewDecFromStr("-1.424742501084004701196565276318876743"),
+		// },
+
+		// THIS TEST CASE FAILS WITH 0.722861284485807159166346369373275311 ERROR
+		// "log_500000(0.0001) = -0.701881216598197542030218906945601429": {
+		// 	initialValue: NewBigDec(500000),
+		// 	base:         NewDecWithPrec(1, 4),
+		// 	// From: https://www.wolframalpha.com/input?i=log_500000%280.0001%29+to+36+digits
+		// 	expected: MustNewDecFromStr("-0.701881216598197542030218906945601429"),
+		// },
+
+		"log_10000(5000000) = 1.674742501084004701196565276318876743": {
+			initialValue: NewBigDec(5000000),
+			base:         NewBigDec(10000),
+			// From: https://www.wolframalpha.com/input?i=log_10000%285000000%29+to+37+digits
+			expected: MustNewDecFromStr("1.674742501084004701196565276318876743"),
+		},
+		"log_0.123456789(1) = 0": {
+			initialValue: OneDec(),
+			base:         MustNewDecFromStr("0.123456789"),
+			expected:     ZeroDec(),
+		},
+		"log_1111(1111) = 1": {
+			initialValue: NewBigDec(1111),
+			base:         NewBigDec(1111),
+			expected:     OneDec(),
+		},
+		// Fails with 0.000000000000000000000000000000007686!!!!!
+		"log_1.123{1024.987654321} = 59.760484327223888489694630378785099461": {
+			initialValue: NewDecWithPrec(1024987654321, 9),
+			base:         NewDecWithPrec(1123, 3),
+			// From: https://www.wolframalpha.com/input?i=log_1.123%281024.987654321%29+to+38+digits
+			expected: MustNewDecFromStr("59.760484327223888489694630378785099461"),
+		},
+		// Fails with 0.000000000000000000000000000000077705!!!!!
+		"log_1.123{912648174127941279170121098210.92821920190204131121} = 594.689327867863079177915648832621538986": {
```

Comment:
I think this is fine. Let's do the following:
- add a test case field `expectedErrorTolerance`
- select the minimal `expectedErrorTolerance` value for each test case